lib/resty/lock.lua

```diff
@@ -25,6 +24,17 @@ local FREE_LIST_REF = 0
 local memo = {}
 if debug then _M.memo = memo end
 
+-- ngx.sleep is disabled when init or init_worker phase, so here need a judgement condition.
+local old_sleep = ngx.sleep
+ngx.sleep = function(s)
+    local phase = ngx.get_phase()
+    if phase == "init" or phase == "init_worker" then
+        os.execute(string.format("sleep %f", s))
```

Comment:
This is very bad. It will block the nginx worker processes for the `init_worker_by_lua*` handlers. Don't do it.

Reply:
Is there any betty way to fix the "API disabled in the context of init_worker_by_lua*" in function 'sleep' error when we use resty.lock at init_worker phase?

Reply:
@807801002 Sure, for init_worker_by_lua*, you could put the whole handler logic into a 0-delay timer, for example.

And there is plan to support yield-able Lua API in the context of `init_by_lua*` this year.

Reply:
I see, I've uesd the "ngx.timer.at" to handle my core logic at init_by_worker phase, but the "resty.lock" is not, let me try to move it into the timer and test that.